resources/js/components/ui/Table/ColumnName.tsx

```diff
@@ -0,0 +1,32 @@
+import {ColumnDef} from "@tanstack/solid-table";
+import {Component, Show} from "solid-js";
+import {Capitalize} from "../Capitalize";
+import {useTable} from "./TableContext";
+
+interface Props {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  def: ColumnDef<any>;
+}
+
+/**
+ * Component displaying the column name, taken from column meta.columnName if present,
+ * otherwise from table meta.translations, or finally from the column id.
+ */
+export const ColumnName: Component<Props> = (props) => {
+  const table = useTable();
+  return (
+    <Show
```

Comment:
Te Show z callbackami wydają się trochę zawiłe. Da się może bardziej wprost, np. jakoś tak

{props.def.meta?.columnName} || <Capitalize text={props.def.meta?.translations?.headers?... ||  props.def.id} />

?

Reply:
Dzięki, powoli zaczynam kumać, choć trochę się miesza. Ale wydaje się że pierwszy i drugi powód to może jednak ta paranoja. 

Jak rozumiem trakowanie wyrażeń z ifami działa tak, że śledzi rzeczy które miały wpływ na aktualny wynik (nie wszystkie możliwe dependencje w innych branchach, bo tego chyba się nie da zrobić w ogólności). Nie trzeba wiedzieć że wyrażenie zależy od `c` póki `a` jest truthy. Najpierw musi się zmienić `a`, żeby zależność na `c` miała znaczenie.

Zobacz moje eksperymenty tutaj: https://playground.solidjs.com/anonymous/baec1d38-2e1e-4eef-bcea-a18a6f82ee40 Zdają się potwierdzać, że powody 1 i 2 nie są problemem i nie wskazują na różnice pomiędzy Show oraz "a ? b : c".

Zresztą sama funkcja Show nie jest jakoś przesadnie magiczna: https://github.com/solidjs/solid/blob/f79ba4d9087d74ef7e55fd0dc336c98ce2ec8526/packages/solid/src/render/flow.ts#L86 Nie całą rozumiem i dochodzą te createMemo, ale chyba powinna się sprowadzać do podobnej własnej funkcji na ifach lub `a ? b : c`. 

Jeśli chodzi o ten trzeci powód, to też nie jestem pewien, ale nie wiem jak to sprawdzać. Na MyComponent to się nie potwierdza, nie jest tworzony wielokrotnie, ale nie wiem jak w przypadku <span>. 

To wszystko ma chyba związek z https://docs.solidjs.com/guides/foundations/understanding-components#why-are-components-different-in-solid Ten link akurat też zaleca Show "to achieve the same result except with better reactivity and optimization", chociaż tych różnic w reactivity nie udało mi się zobaczyć w playground.

W kodzie oczywiście ma sens zostawić Show skoro tak jest standardowo. Ja tak dla własnego zrozumienia drążę.

Reply:
No nieźle to poanalizowałeś, i wygląda że co do 1. to faktycznie moja paranoja, co do 2. to wygląda że on usuwa dependencje jeżeli ostatnia ewaluacja do pewnych sygnałów nie sięgnęła, myślałem że tego nie robi, coś tam testowałem ale nie wykryłem tego wtedy. Nie wiem czy czuję się całkiem uspokojony :)

W fn2 masz `<span>b()</span>` a powinno być `<span>{b()}</span>`, wtedy widać że zmiana a powoduje chyba tworzenie spana od nowa - albo co najmniej loguje się tamta linia. Natomiast co do tworzenia komponentu to najwyraźniej odbywa się jakaś magia wyższego poziomu gdy widzi ten ternary. Zobacz że jak to co masz w Result 4, 5 i 6 owiniesz całe w IIFE, to ten komponent się tworzy od nowa przy zmianie a (dla 5 i 6, a dla 4 nie).

W efekcie zdaje mi się, że ten solid jest lepiej po prostu zoptymalizowany niż sądziłem, ale nadal pewne rzeczy są ryzykowne. Jest też linter który narzeka na różne rzeczy które próbujesz robić źle w temacie reaktywności, no ale nie na wszystkie więc nadal trzeba być czujnym: https://github.com/solidjs-community/eslint-plugin-solid/blob/main/docs/reactivity.md

Z komponentów do flow jest jeszcze For i Index które machają tymi sygnałami bardziej nawet niż Show, i większość swoich bugów zrobiłem właśnie w Index'ie, którego kod nie jest aż taki prosty, i nie wszystko udało mi się wyjaśnić, ale za to mniej więcej ustaliłem jak pisać żeby raczej działało.

Czyli pomogłem ci zrozumieć to lepiej niż ja rozumiałem :) spoko.

Reply:
Aha, a co do b versus b(), to niby funkcja działa w takim miejscu, ale gdzieś chyba pisali że to nie jest polecane, i należy ją jednak wywołać samemu, ale nie pamiętam szczegółów. No i pewnie zostanie kompatybilność do końca świata.